### ISSUE TITLE
Support for Links to Headings

### ISSUE THREAD

**Issue by 1dancook** [kortina/vscode-markdown-notes#34]
There is a todo note in the readme about support for linking to headings. I did not see an issue for this (closed or open) so I'm making one.

Please support links to headings. Syntax:

`[[file.md#heading]]`

It could be good if you could peek the heading location as well.

Thanks!

**Comment by kortina** [comment]
Agree, this would be great! Shouldn't be terribly hard to do, and would also be a good oppy to clear up any collisions between wiki-link and tag regexes.

**Comment by stemcc** [comment]
Do you have any suggestions for getting started on a PR for this feature?

**Comment by lukesmurray** [comment]
For markdown files the symbol tree consists of the headers in the document. I think if you execute the command [`vscode.executeDocumentSymbolProvider`](https://code.visualstudio.com/api/references/commands) you could get a list of headers for any markdown document.

Anyway this isn't a solution in general but at least you don't have to write any parsing code to find the headers. vscode does all the work for you automatically since the markdown language server is built in.

**Comment by astrelochnik** [comment]
+1 for this idea.

**Comment by rshalaev** [comment]
+1 from me as well. Linking between notes been a weak point of several markdown editors, and it's holding up use-cases where you have extensive documentation in markdown and want to link up to a specific section.

**Comment by Askhento** [comment]
Any progress on this one?
